updated SecurityConfig.java to include making device_manager page res…

Bsbuckne opened the pull request:
…tricted without logging in

Commit:
updated SecurityConfig.java to include making device_manager page restricted without logging in

Bsbuckne commented:
This needs to be merged before I can test it on master, it works fine on the device manager branch. Check it out and if it's cool merge it and I'll QA it tomorrow

cit280superepicness committed:
Fixed where header links were showing up whether user had ROLE_ADMIN or ROLE_USER roles.

cit280superepicness committed:
Moved index.html to templates folder, moved index.html styling to custom css stylesheet, created page fragment in header.html for index.html with spring security controlling header links

cit280superepicness committed:
Finished header for index.html. Fixed issue  with text being at slightly different heights in header

cit280superepicness committed:
Streamlined comments in bootstrap.custom.css

cit280superepicness commented:
Conflicts with master resolved.

jrmullen commented on src/main/resources/templates/index.html:
```diff
@@ -0,0 +1,52 @@
+<!DOCTYPE html>
+<html xmlns:th="http://www.thymeleaf.org" xmlns:sec="http://www.thymeleaf.org/thymeleaf-extras-springsecurity4">
+<head>
+    <!-- Theme Made By www.w3schools.com - No Copyright -->
```
remove comment

jrmullen commented on src/main/resources/templates/index.html:
```diff
@@ -0,0 +1,52 @@
+<div th:include="fragments/header :: indexHeader"></div>
+
+
+<!-- First Container -->
```
remove comment

jrmullen commented on src/main/resources/templates/index.html:
```diff
@@ -0,0 +1,52 @@
+    <img src="http://i67.tinypic.com/2d0uw8.jpg" class="img-responsive" style="display:inline" alt="camera" width="450" height="450" />
+</div>
+
+<!-- Second Container (Grid) -->
```
remove comment

jrmullen commented on src/main/java/com/watchdog/config/SecurityConfig.java:
```diff
@@ -52,8 +53,8 @@ public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception
-                .authoritiesByUsernameQuery("select user.USER_EMAIL, permissions.ROLE from user, permissions "
-                                                            + "where user.USER_EMAIL = ?");
+                .authoritiesByUsernameQuery("select user.USER_EMAIL, permissions.ROLE from user, permissions " +
+                        "where permissions.PERMISS_ID = user.PERMISS_ID and user.USER_EMAIL = ?");
```
New query has been tested, correct?

cit280superepicness commented:
Yes, the new query has been tested. I am not able to get to my computer to
remove the comments till much later today so if you could remove the
comments yourself or just go ahead and merge so that Briana can watch test
it today like she mentioned y, that would be great.

On Oct 24, 2016 9:53 AM, "Jeremy Mullen" notifications@github.com wrote:

> ## _@jrmullen_ commented on this pull request.
> 
> In src/main/java/com/watchdog/config/SecurityConfig.java
> https://github.com/jrmullen/Watchdog/pull/44#pullrequestreview-5452726:
> 
> > @@ -52,8 +53,8 @@ public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception
> >                  .jdbcAuthentication().dataSource(dataSource)
> >                  .passwordEncoder(passwordEncoder())
> >                  .usersByUsernameQuery("SELECT USER_EMAIL, USER_PASSWORD, enabled FROM user WHERE USER_EMAIL = ?")
> > -                .authoritiesByUsernameQuery("select user.USER_EMAIL, permissions.ROLE from user, permissions "
> > -                                                            + "where user.USER_EMAIL = ?");
> > -                .authoritiesByUsernameQuery("select user.USER_EMAIL, permissions.ROLE from user, permissions " +
> > -                        "where permissions.PERMISS_ID = user.PERMISS_ID and user.USER_EMAIL = ?");
> 
> New query has been tested, correct?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> https://github.com/jrmullen/Watchdog/pull/44#pullrequestreview-5452726,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AIegPUDlakCZdrCX4v9jQbqoG2fBfKwvks5q3Lg9gaJpZM4KeSX0
> .

cit280superepicness commented:
I will look into it. I also found a workaround for
sec:authorize="hasRole('ROLE_ADMIN')" not working. I am going to see if I
can get it working on the device_manager branch. I will resolve the merge
conflicts as well.

I'll let you know if I am successful/when I am done.

On Sun, Oct 23, 2016 at 8:10 PM, Bsbuckne notifications@github.com wrote:

> This needs to be merged before I can test it on master, it works fine on
> the device manager branch. Check it out and if it's cool merge it and I'll
> QA it tomorrow
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> https://github.com/jrmullen/Watchdog/pull/44#issuecomment-255624557, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AIegPZR0Vtb7JOs4JcywbsEHYeQQET4dks5q2_eEgaJpZM4KeSX0
> .

cit280superepicness commented:
The header is now only showing content I should for users with role_admin
and role_user.

On Oct 24, 2016 12:47 PM, "Renee Wise" rlwise@oakland.edu wrote:

> Yes, the new query has been tested. I am not able to get to my computer to
> remove the comments till much later today so if you could remove the
> comments yourself or just go ahead and merge so that Briana can watch test
> it today like she mentioned y, that would be great.
> 
> On Oct 24, 2016 9:53 AM, "Jeremy Mullen" notifications@github.com wrote:
> 
> > ## _@jrmullen_ commented on this pull request.
> > 
> > In src/main/java/com/watchdog/config/SecurityConfig.java
> > https://github.com/jrmullen/Watchdog/pull/44#pullrequestreview-5452726:
> > 
> > > @@ -52,8 +53,8 @@ public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception
> > >                  .jdbcAuthentication().dataSource(dataSource)
> > >                  .passwordEncoder(passwordEncoder())
> > >                  .usersByUsernameQuery("SELECT USER_EMAIL, USER_PASSWORD, enabled FROM user WHERE USER_EMAIL = ?")
> > > -                .authoritiesByUsernameQuery("select user.USER_EMAIL, permissions.ROLE from user, permissions "
> > > -                                                            + "where user.USER_EMAIL = ?");
> > > -                .authoritiesByUsernameQuery("select user.USER_EMAIL, permissions.ROLE from user, permissions " +
> > > -                        "where permissions.PERMISS_ID = user.PERMISS_ID and user.USER_EMAIL = ?");
> > 
> > New query has been tested, correct?
> > 
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > https://github.com/jrmullen/Watchdog/pull/44#pullrequestreview-5452726,
> > or mute the thread
> > https://github.com/notifications/unsubscribe-auth/AIegPUDlakCZdrCX4v9jQbqoG2fBfKwvks5q3Lg9gaJpZM4KeSX0
> > .

Bsbuckne commented:
I already fixed the device manager issue with the roles, I just haven't pushed it yet

> On Oct 25, 2016, at 2:41 PM, Renee notifications@github.com wrote:
> 
> I will look into it. I also found a workaround for
> sec:authorize="hasRole('ROLE_ADMIN')" not working. I am going to see if I
> can get it working on the device_manager branch. I will resolve the merge
> conflicts as well.
> 
> I'll let you know if I am successful/when I am done.
> 
> On Sun, Oct 23, 2016 at 8:10 PM, Bsbuckne notifications@github.com wrote:
> 
> > This needs to be merged before I can test it on master, it works fine on
> > the device manager branch. Check it out and if it's cool merge it and I'll
> > QA it tomorrow
> > 
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > https://github.com/jrmullen/Watchdog/pull/44#issuecomment-255624557, or mute
> > the thread
> > https://github.com/notifications/unsubscribe-auth/AIegPZR0Vtb7JOs4JcywbsEHYeQQET4dks5q2_eEgaJpZM4KeSX0
> > .
> > 
> > —
> > You are receiving this because you modified the open/close state.
> > Reply to this email directly, view it on GitHub, or mute the thread.

cit280superepicness commented:
Briana, a bunch of fixes related to roles were already pushed to the main
branch this morning. what specific issue with roles are you referring to?

Bsbuckne commented:
I was still able to see the link to and get to the device_manager when logged in as a user. I fixed it to where that wasn't the case anymore

> On Oct 25, 2016, at 5:55 PM, Renee notifications@github.com wrote:
> 
> Briana, a bunch of fixes related to roles were already pushed to the main
> branch this morning. what specific issue with roles are you referring to?
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub, or mute the thread.

cit280superepicness commented:
I fixed this and tested it on my system before and then pushed it to the
device manager branch. Unless someone saved over what I pushed before the
device manager was merged with master and that is the issue, did you try
doing "git pull origin master" or "git pull origin device_manager" to get
the most recent changes? Yo should be able to pull all the fixes.

Also, did you import the most recent version of the db file I uploaded
today? I repopulated the users table a few days ago to  have 1/2 the users
be admins and be normal users. Then today I made another change to the
script, I made it by default so that when you register as a  new user in
Watchdog, the user is a regular user and not an admin. If you are having
issues with a newly registered us er being able to access the
device_manager, not having the most recent script is why.

On Oct 25, 2016 5:59 PM, "Bsbuckne" notifications@github.com wrote:

> I was still able to see the link to and get to the device_manager when
> logged in as a user. I fixed it to where that wasn't the case anymore
> 
> > On Oct 25, 2016, at 5:55 PM, Renee notifications@github.com wrote:
> > 
> > Briana, a bunch of fixes related to roles were already pushed to the main
> > branch this morning. what specific issue with roles are you referring to?
> > —
> > You are receiving this because you modified the open/close state.
> > Reply to this email directly, view it on GitHub, or mute the thread.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> https://github.com/jrmullen/Watchdog/pull/44#issuecomment-256189498, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AIegPbLRW-z_2fDPIxkGBnBhAyYl6ax4ks5q3nu-gaJpZM4KeSX0
> .

cit280superepicness commented:
Hey Briana,

So I just looked at the device manager branch and everything is fine. Sorry
for the confusion on my part. I also see now you were right about that line
to restrict access based on user role for device_manager missing in the
SecurityConfig file.Thanks for noticing that and adding it!

Renee

On Tue, Oct 25, 2016 at 6:19 PM, Renee Wise rlwise@oakland.edu wrote:

> I fixed this and tested it on my system before and then pushed it to the
> device manager branch. Unless someone saved over what I pushed before the
> device manager was merged with master and that is the issue, did you try
> doing "git pull origin master" or "git pull origin device_manager" to get
> the most recent changes? Yo should be able to pull all the fixes.
> 
> Also, did you import the most recent version of the db file I uploaded
> today? I repopulated the users table a few days ago to  have 1/2 the users
> be admins and be normal users. Then today I made another change to the
> script, I made it by default so that when you register as a  new user in
> Watchdog, the user is a regular user and not an admin. If you are having
> issues with a newly registered us er being able to access the
> device_manager, not having the most recent script is why.
> 
> On Oct 25, 2016 5:59 PM, "Bsbuckne" notifications@github.com wrote:
> 
> > I was still able to see the link to and get to the device_manager when
> > logged in as a user. I fixed it to where that wasn't the case anymore
> > 
> > > On Oct 25, 2016, at 5:55 PM, Renee notifications@github.com wrote:
> > > 
> > > Briana, a bunch of fixes related to roles were already pushed to the
> > > main
> > > branch this morning. what specific issue with roles are you referring
> > > to?
> > > —
> > > You are receiving this because you modified the open/close state.
> > > Reply to this email directly, view it on GitHub, or mute the thread.
> > 
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > https://github.com/jrmullen/Watchdog/pull/44#issuecomment-256189498,
> > or mute the thread
> > https://github.com/notifications/unsubscribe-auth/AIegPbLRW-z_2fDPIxkGBnBhAyYl6ax4ks5q3nu-gaJpZM4KeSX0
> > .

Bsbuckne commented:
I made my fixes to the device manager after we presented today and no I haven't imported your most recent script. I haven't touched our project since we left class earlier. I'll do another pull on Thursday. I'll be working on my other two class projects tonight and tomorrow 

> On Oct 25, 2016, at 6:19 PM, Renee notifications@github.com wrote:
> 
> I fixed this and tested it on my system before and then pushed it to the
> device manager branch. Unless someone saved over what I pushed before the
> device manager was merged with master and that is the issue, did you try
> doing "git pull origin master" or "git pull origin device_manager" to get
> the most recent changes? Yo should be able to pull all the fixes.
> 
> Also, did you import the most recent version of the db file I uploaded
> today? I repopulated the users table a few days ago to have 1/2 the users
> be admins and be normal users. Then today I made another change to the
> script, I made it by default so that when you register as a new user in
> Watchdog, the user is a regular user and not an admin. If you are having
> issues with a newly registered us er being able to access the
> device_manager, not having the most recent script is why.
> 
> On Oct 25, 2016 5:59 PM, "Bsbuckne" notifications@github.com wrote:
> 
> > I was still able to see the link to and get to the device_manager when
> > logged in as a user. I fixed it to where that wasn't the case anymore
> > 
> > > On Oct 25, 2016, at 5:55 PM, Renee notifications@github.com wrote:
> > > 
> > > Briana, a bunch of fixes related to roles were already pushed to the main
> > > branch this morning. what specific issue with roles are you referring to?
> > > —
> > > You are receiving this because you modified the open/close state.
> > > Reply to this email directly, view it on GitHub, or mute the thread.
> > 
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > https://github.com/jrmullen/Watchdog/pull/44#issuecomment-256189498, or mute
> > the thread
> > https://github.com/notifications/unsubscribe-auth/AIegPbLRW-z_2fDPIxkGBnBhAyYl6ax4ks5q3nu-gaJpZM4KeSX0
> > .
> > 
> > —
> > You are receiving this because you modified the open/close state.
> > Reply to this email directly, view it on GitHub, or mute the thread.
